drivers/gpio/gpio_sam0.c

```diff
@@ -88,23 +172,103 @@ static int gpio_sam0_read(struct device *dev, int access_op, u32_t pin,
 	return 0;
 }
 
+static int gpio_sam0_manage_callback(struct device *dev,
+				     struct gpio_callback *callback, bool set)
+{
+	struct gpio_sam0_data *data = dev->driver_data;
+
+	_gpio_manage_callback(&data->cb, callback, set);
+
+	return 0;
+}
+
+static int gpio_sam0_enable_callback(struct device *dev, int access_op,
+				     u32_t pin)
+{
+	struct gpio_sam0_data *data = dev->driver_data;
+
+	if (access_op != GPIO_ACCESS_BY_PIN) {
+		return -ENOTSUP;
+	}
+
+	data->cb_pins |= 1 << pin;
+
+	return 0;
+}
+
+static int gpio_sam0_disable_callback(struct device *dev, int access_op,
+				      u32_t pin)
+{
+	struct gpio_sam0_data *data = dev->driver_data;
+
+	if (access_op != GPIO_ACCESS_BY_PIN) {
+		return -ENOTSUP;
+	}
+
+	data->cb_pins &= ~(1 << pin);
+
+	return 0;
+}
+
+static u32_t gpio_sam0_get_pending_int(struct device *dev) { return 0; }
+
+static void gpio_sam0_isr(struct device *dev)
+{
+	struct gpio_sam0_data *data = dev->driver_data;
+	u32_t flags = EIC->INTFLAG.reg;
+
+	/* Acknowledge all interrupts */
+	EIC->INTFLAG.reg = flags;
+
+	flags &= data->cb_pins;
+	if (flags != 0) {
+		_gpio_fire_callbacks(&data->cb, dev, flags);
+	}
+}
+
+int gpio_sam0_init(struct device *dev)
+{
+	/* Enable the EIC clock in PM */
+	PM->APBAMASK.bit.EIC_ = 1;
+
+	/* Enable the GCLK */
+	GCLK->CLKCTRL.reg = GCLK_CLKCTRL_ID_EIC | GCLK_CLKCTRL_GEN_GCLK0 |
+			    GCLK_CLKCTRL_CLKEN;
+
+	/* Reset the EIC */
+	EIC->CTRL.reg = EIC_CTRL_SWRST;
+	gpio_sam0_sync_eic();
+
+	/* Route and enable the interrupt.  This is safe as nothing
+	 * happens until an individual channel is unmasked.
+	 */
+	IRQ_CONNECT(EIC_IRQn, 0, gpio_sam0_isr, DEVICE_GET(gpio_sam0_0), 0);
```

Comment:
We shouldn't force interrupt priority to a specific value, regardless of the fact how well we write interrupt handler. This is a sensitive subject. Let's have interrupt priority configurable via DTS (or Kconfig).

Reply:
Ack, will move to DTS.

Reply:
Done.